### PR TITLE
Add sampled input to DistNeighborSampler._sample_from_nodes

### PR DESCRIPTION
**Pull request by kmontemayor2-sc**
I want to do this as I'd like to be able to maintain a full copy of the inputs to the sampling function - and AFAICT we do not currently store this anywhere.

My usecase here is that I have some specially constructed tensor which contains labels to sample against, and I'd like to be able to identify the "label nodes". 

The tensor looks something like: `[<src_node>, <padding>, <positive_labels>, ...]`, and since this `.node` and `.batch` get deduped somewhere (idk where) I cannot reconstruct the labels per src node otherwise.

### CODE REVIEW

File: graphlearn_torch/python/loader/transform.py [hunk 1]
```diff
@@ -113,21 +113,24 @@ def to_hetero_data(
...
   input_type = hetero_sampler_out.input_type
   if isinstance(hetero_sampler_out.metadata, dict):
-    # if edge_dir == 'out', we need to reverse the edge type
-    res_edge_type = reverse_edge_type(input_type) if edge_dir == 'out' else input_type
     for k, v in hetero_sampler_out.metadata.items():
-      if k == 'edge_label_index':
-        if edge_dir == 'out':
-          data[res_edge_type]['edge_label_index'] = \
-            torch.stack((v[1], v[0]), dim=0)
+      if isinstance(input_type, tuple):
```
Comment (kmontemayor2-sc) [inline graphlearn_torch/python/loader/transform.py:117]: Making this change as `input_type` can also be a `NodeType` or `None`, both of which will throw if passed into `reverse_edge_type`.
